File: src/main/java/com/ck/test/TestSubject.java
package com.ck.test;

import java.util.ArrayList;

// Test Subject 1: Simple Class
class Person {
    private String name;
    private int age;

    public void setName(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    // Uses 'age'
    public void setAge(int age) {
        this.age = age;
    }

    // Uses nothing (LCOM impact)
    public void doNothing() {
        System.out.println("Nothing");
    }
}

// Test Subject 2: Inheritance
class Employee extends Person {
    private double salary;

    public void setSalary(double salary) {
        this.salary = salary;
    }
}

// Test Subject 3: Coupling
class Department {
    private Manager manager; // Coupling 1
    private ArrayList<Employee> employees; // Coupling 2 (ArrayList, Employee)

    public void addEmployee(Employee e) { // Method param
        // ...
    }

    public Manager getManager() { // Return type
        return manager;
    }
}

// Test Subject 4: Another class
class Manager extends Employee {
    // ...
}

File: CK-Analyser-TS/sample/TestSubject.ts
// Test Subject
export class Person {
    private name: string;
    private age: number;

    constructor(name: string, age: number) {
        this.name = name;
        this.age = age;
    }

    public getName(): string {
        return this.name;
    }

    public getAge(): number {
        return this.age;
    }

    // LCOM: uses age
    public setAge(a: number) {
        this.age = a;
    }

    // LCOM: uses nothing
    public doNothing() {
        console.log("Nothing");
    }
}

export class Employee extends Person {
    private salary: number;

    constructor(name: string, age: number, salary: number) {
        super(name, age);
        this.salary = salary;
    }

    public getSalary(): number {
        return this.salary;
    }
}

export class Manager extends Employee {
    public manage() {
        console.log("Managing...");
    }
}
